packages/vite/src/node/server/hmr.ts

```diff
@@ -793,6 +793,7 @@ function propagateUpdate(
     // #3716, #3913
     // For a non-CSS file, if all of its importers are CSS files (registered via
     // PostCSS plugins) it should be considered a dead end and force full reload.
+    // TODO
     if (
       !isCSSRequest(node.url) &&
       [...node.importers].every((i) => isCSSRequest(i.url))
```

Comment:
With this PR, changing any file referenced by CSS will cause a full reload because of this code (slightly related: this code is the cause of #9512). That'll probably degrade the UX quite a bit.

For this reason, I'm considering to only handle inlined SVGs for now (in a different PR).